app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+        int pointerIndex = motionEvent.getPointerId(motionEvent.getActionIndex());
+        int pointerCount = motionEvent.getPointerCount();
+
+        switch (motionEvent.getActionMasked()){
+            case MotionEvent.ACTION_DOWN:
+            case MotionEvent.ACTION_POINTER_DOWN:
+
+                recodeFirstHistorical(motionEvent);
+                actionDownTime[pointerIndex] = upTime;
+
+                pointerMode = pointerCount-1;
+                switch (pointerMode){
+                    case 1: // One -> Two
+                        resetOnePointer();
+                        break;
+                    case 2: // Two -> Triple
+                        isRight = false;
+                        resetDoublePointer();
+                        break;
+                }
+
+                break;
+            case MotionEvent.ACTION_BUTTON_PRESS:
+            case MotionEvent.ACTION_BUTTON_RELEASE:
+                break;
+            case MotionEvent.ACTION_MOVE:
+                getRelative(motionEvent);
+                break;
+
+            case MotionEvent.ACTION_UP:
+                switch (pointerMode){
+                    case 0:
+                        break;
+                    case 1:
+                        resetDoublePointer();
+                        break;
+                    case 2:
+                        isRight = false;
+                        resetTriplePointer();
+                        break;
+                }
+
+                pointerMode = 0;
+                actionUpTime[pointerIndex] = upTime;
+                break;
+
+            case MotionEvent.ACTION_POINTER_UP:
+                switch (pointerMode){
+                    case 1:  // Double -> Reset
+                        resetDoublePointer();
+                        break;
+                    case 2: // Triple -> Reset
+                        isRight = false;
+                        resetTriplePointer();
+                        break;
+                }
+
+                pointerMode = -1; // Block
+                isRight = false;
+                actionUpTime[pointerIndex] = upTime;
+                break;
+
+            default:
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                    System.out.println("New Action!! : "+MotionEvent.actionToString(motionEvent.getAction())+
+                            " : "+motionEvent.getAction());
+                }
+                break;
+
+        }
+    }
+
+    private final float[] firstX = new float[4];
+    private final float[] firstY = new float[4];
+    private final float[] historicalAxisX = new float[4];
+    private final float[] historicalAxisY = new float[4];
+    private final float[] lastX = new float[4];
+    private final float[] lastY = new float[4];
+
+    private final float[] axis_Relative_X = new float[4];
+    private final float[] axis_Relative_Y = new float[4];
+
+    private void recodeFirstHistorical(MotionEvent motionEvent){
```

Comment:
I assume this should be `record` instead of `recode`?

Reply:
It's a typo. I'll fix it.